core/modules/qdisp/Executive.h

```diff
@@ -75,7 +75,7 @@ class Executive {
     /// Construct an Executive.
     /// If c->serviceUrl == Config::getMockStr(), then use XrdSsiServiceMock
     /// instead of a real XrdSsiService
-    Executive(Config::Ptr c, std::shared_ptr<MessageStore> ms);
+    static Executive::Ptr newExecutive(Config::Ptr const& c, std::shared_ptr<MessageStore> const& ms);
```

Comment:
For enabled_shared_from_this to work properly, the object must be created as a shared pointer. There's a problem if someone does
<code>
    Executive* exec = new Executive(...);
<code>
So making a private constructor and the factory function provides a reasonable amount of protection. Unfortunately, std::make_shared really doesn't like private constructors, the factory function just uses new.